rust/processor/migrations/2024-05-22-200847_add_v1_migration_views/up.sql

```diff
@@ -0,0 +1,319 @@
+-- Your SQL goes here
+
+
+-- Create the schema
+CREATE SCHEMA IF NOT EXISTS legacy_migration_v1;
+
+-- Replace `move_resources` with account transactions
+CREATE OR REPLACE VIEW legacy_migration_v1.move_resources AS
+    SELECT *
+    FROM account_transactions at2
+  ;
+-- account_transactions already has index on transaction_version
+
+-- replace `coin_activities` with `fungible_asset_activities`
+CREATE OR REPLACE VIEW legacy_migration_v1.coin_activities AS
+    SElECT
+        transaction_version,
+        owner_address as event_account_address,
+        -- these two below are mildly concerning
+        0 as event_creation_number,
+        0 as event_sequence_number,
+        owner_address,
+        asset_type AS coin_type,
+        amount,
+        "type" AS activity_type,
+        is_gas_fee,
+        is_transaction_success,
+        entry_function_id_str,
+        block_height,
+        transaction_timestamp,
+        inserted_at,
+        event_index,
+        gas_fee_payer_address,
+        storage_refund_amount,
+    FROM public.fungible_asset_activities
+    WHERE token_standard = 'v1'
+  ;
+CREATE INDEX lm1_ca_ct_a_index ON public.fungible_asset_activities USING btree (asset_type, amount)
+CREATE INDEX lm1_ca_ct_at_a_index ON public.fungible_asset_activities USING btree (asset_type, "type", amount)
+CREATE INDEX lm1_ca_oa_ct_at_index ON public.fungible_asset_activities USING btree (owner_address, asset_type, "type", amount)
+CREATE INDEX lm1_ca_oa_igf_index ON public.fungible_asset_activities USING btree (owner_address, is_gas_fee)
+
+-- replace `coin_balances` with `fungible_asset_balances`
+CREATE OR REPLACE VIEW legacy_migration_v1.coin_balances AS
+    SELECT
+        transaction_version,
+        owner_address,
+        -- this is mainly for hashing the coin type for primary key
+        encode(sha256(asset_type::bytea), 'hex') as coin_type_hash,
+        asset_type as coin_type,
+        amount,
+        transaction_timestamp,
+        inserted_at
+    FROM public.fungible_asset_balances
+    WHERE token_standard = 'v1'
+    ;
+CREATE INDEX lm1_cb_tv_oa_ct_index ON public.fungible_asset_balances USING btree (transaction_version, owner_address, asset_type)
+
+-- replace `current_coin_balances` with `current_fungible_asset_balances`
+CREATE OR REPLACE VIEW legacy_migration_v1.current_coin_balances AS
+    SELECT
+        owner_address,
+        substring(asset_type, 64) as coin_type_hash,
+        asset_type as coin_type,
+        amount,
+        transaction_version as last_transaction_version,
+        transaction_timestamp as last_transaction_timestamp,
+        inserted_at
+    FROM public.current_fungible_asset_balances
+    WHERE token_standard = 'v1'
+    ;
+CREATE INDEX lm1_ccb_ct_a_index ON public.current_fungible_asset_balances USING btree (asset_type, amount)
```

Comment:
surprising this didn't already exist

Reply:
This index is also used very infrequently. Usually queries have owner